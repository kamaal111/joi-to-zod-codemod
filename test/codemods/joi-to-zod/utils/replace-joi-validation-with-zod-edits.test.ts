import { test, expect } from 'vitest';
import { Lang, parseAsync } from '@ast-grep/napi';

import replaceJoiValidationWithZodEdits from '../../../../src/codemods/joi-to-zod/utils/replace-joi-validation-with-zod-edits';

test('replaces Joi required with optional edits', async () => {
  const source = `
import Joi from 'joi';

export const employee = Joi.object().keys({
    name: Joi.string().regex(/^[a-z0-9]+$/).min(3).max(30).required()
})
`;
  const ast = await parseAsync(Lang.TypeScript, source);
  const root = ast.root();
  const edits = replaceJoiValidationWithZodEdits(root, {
    primitive: '*',
    validationTargetKey: 'required()',
    zodValidation: 'optional()',
  });

  const updatedSource = root.commitEdits(edits);

  expect(edits.length).toBe(1);
  expect(updatedSource).not.toContain('required');
  expect(updatedSource).toContain('optional');
  expect(updatedSource).toMatchSnapshot();
});
