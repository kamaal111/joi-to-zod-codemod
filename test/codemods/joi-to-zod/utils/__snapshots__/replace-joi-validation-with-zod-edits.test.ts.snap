// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`replaces Joi required with optional edits 1`] = `
"import Joi from 'joi';

export const employee = Joi.object().keys({
    name: Joi.string().regex(/^[a-z0-9]+$/).min(3).max(30).optional()
})
"
`;
